airflow/api_connexion/endpoints/backfill_endpoint.py

```diff
@@ -0,0 +1,181 @@
+    session.commit()
+    return br
+
+
+def _has_access(dag_ids, methods):
+    if get_auth_manager().filter_permitted_dag_ids(dag_ids=dag_ids, methods=methods, user=g.user):
+        return True
+    return False
+
+
+@security.requires_access_dag("GET")
+@action_logging
+@provide_session
+def list_backfills(dag_id, session):
+    backfills = session.scalars(select(Backfill).where(Backfill.dag_id == dag_id)).all()
+    obj = BackfillCollection(
+        backfills=backfills,
+        total_entries=len(backfills),
+    )
+    serialized = backfill_collection_schema.dump(obj)
+    return serialized
+
+
+@security.requires_access_dag("GET")
+@action_logging
+@provide_session
+def pause_backfill(backfill_id, session):
+    br = session.get(Backfill, backfill_id)
+    if not _has_access(dag_ids=[br.dag_id], methods="PUT"):
+        raise PermissionDenied()
+    if br.completed_at:
+        raise Conflict("Backfill is already completed.")
+    if br.is_paused is False:
+        br.is_paused = True
+    session.commit()
+    return backfill_schema.dump(br)
+
+
+@security.requires_access_dag("GET")
+@action_logging
+@provide_session
+def unpause_backfill(backfill_id, session):
+    br = session.get(Backfill, backfill_id)
+    if not _has_access(dag_ids=[br.dag_id], methods="PUT"):
+        raise PermissionDenied()
+    if br.completed_at:
+        raise Conflict("Backfill is already completed.")
+    if br.is_paused:
+        br.is_paused = False
+    session.commit()
+    return backfill_schema.dump(br)
+
+
+@security.requires_access_dag("GET")
+@action_logging
+@provide_session
+def cancel_backfill(backfill_id, session):
+    br: Backfill = session.get(Backfill, backfill_id)
+    if not _has_access(dag_ids=[br.dag_id], methods="PUT"):
+        raise PermissionDenied()
+
+    if br.completed_at is not None:
+        raise Conflict("Backfill is already completed.")
+
+    br.completed_at = timezone.utcnow()
+
+    # first, pause
+    if not br.is_paused:
+        br.is_paused = True
+    session.commit()
+    return backfill_schema.dump(br)
+
+
+@security.requires_access_dag("GET")
+@action_logging
+@provide_session
+def get_backfill(backfill_id: int, session: Session = NEW_SESSION):
+    backfill = session.get(Backfill, backfill_id)
+    if backfill:
+        if not _has_access([backfill.dag_id], methods=["GET"]):
+            raise PermissionDenied()
+        return backfill_schema.dump(backfill)
+    raise NotFound("Backfill not found")
+
+
+@security.requires_access_dag("PUT")
+@action_logging
+def create_backfill(
+    dag_id: str,
+    from_date: str,
+    to_date: str,
+    max_active_runs: int = 10,
+    reverse: bool = False,
+    dag_run_conf: dict | None = None,
+) -> APIResponse:
+    backfill_obj = _create_backfill(
+        dag_id=dag_id,
+        from_date=from_date,
+        to_date=to_date,
+        max_active_runs=max_active_runs,
+        reverse=reverse,
+        dag_run_conf=dag_run_conf,
+    )
+    if not backfill_obj:
```

Comment:
Can this even be falsy?

Reply:
this works.  it is equiv to `if backfill_obj is not None`

Reply:
Doesn't `_create_backfill` always return a `Backfill` though?

Reply:
Ah yes, right now it does.  I removed the more complete logic where e.g. we check if there's other active backfills (was going to do a followup with the more complete logic) and if there is we don't create.

But you can see that logic here in the commit that takes it out.
https://github.com/apache/airflow/pull/42455/commits/eacf3b31da460c1c5bb1a3aa6dac643993e4e1dd#diff-3aab1eebde344201c4cf2e609a12eab3417c12aba88f54839c12b2c43007b277L72-L73

It might be better to raise an exception there but 🤷

If you think it's best to remove that if now, not a problem.

Reply:
and done, just to make it easy